src/Components/Aspire.Azure.Messaging.EventHubs/EventHubsComponent.cs

```diff
@@ -101,4 +105,41 @@ protected static string GetNamespaceFromSettings(AzureMessagingEventHubsSettings
                 $"the '{configurationSectionName}' configuration section.");
         }
     }
+
+    protected sealed class AzureEventHubHealthCheck : IHealthCheck, IAsyncDisposable
+    {
+        private readonly EventHubProducerClient _probe;
+
+        public AzureEventHubHealthCheck(TSettings settings)
+        {
+            var probeOptions = new EventHubProducerClientOptions
+            {
+                Identifier = $"AspireEventHubHealthCheck-{settings.EventHubName}"
+            };
+
+            // At this point, the connection string or credential should already be validated by the AddClient call
+            _probe = !string.IsNullOrEmpty(settings.ConnectionString) ?
```

Comment:
There is another case that I am now handling in https://github.com/dotnet/aspire/pull/4209 where the EventHub is not part of the connection string but defined in the settings

https://github.com/dotnet/aspire/pull/4209/files#diff-55b0ddc0e2f1aec7c2af54c1f06af30b30ad2aaf09a64c725e209ae393d87eafR54-R57

Reply:
@sebastienros Is there an action item for me here? I'm not sure.